### ISSUE TITLE
pyshark cannot handle co-related data, canot parse properly.

### ISSUE THREAD

**Issue by srinivasmiskin** [KimiNewt/pyshark#183]
Hi Tried Pyshark on Python3.

below is the original data from pcap:

Cisco NetFlow/IPFIX
    Version: 9
    Count: 1
    SysUptime: 18716.1536247816 seconds
    Timestamp: Mar 14, 2017 19:16:30.000000000 IST
        CurrentSecs: 1489499190
    FlowSequence: 2366
    SourceId: 20
    FlowSet 1
        FlowSet Id: Data Template (V9) (0)
        FlowSet Length: 92
        Template (Id = 256, Count = 21)
            Template Id: 256
            Field Count: 21
            Field (1/21): IP_SRC_ADDR
                Type: IP_SRC_ADDR (8)
                Length: 4
            Field (2/21): IP_DST_ADDR
                Type: IP_DST_ADDR (12)
                Length: 4
            Field (3/21): IP_TOS
                Type: IP_TOS (5)
                Length: 1
.
.

Decoded data:

Layer CFLOW:
	Count: 1
	CurrentSecs: 1489499190
	FlowSet 1
	FlowSequence: 2366
	Field Count: 21
	Version: 9
	FlowSet Id: Data Template (V9) (0)
	FlowSet Length: 92
	SourceId: 20
	SysUptime: 18716.1536247816 seconds
	Timestamp: Mar 14, 2017 19:16:30.000000000 IST
	Type: IP_SRC_ADDR (8)
	Length: 4
	Template Id: 256
	Template (Id = 256, Count = 21)
	Field (1/21): IP_SRC_ADDR
	Field (2/21): IP_DST_ADDR
	Field (3/21): IP_TOS
	Field (4/21): PROTOCOL
	Field (5/21): L4_SRC_PORT
	Field (6/21): L4_DST_PORT
	Field (7/21): ICMP_TYPE
	Field (8/21): INPUT_SNMP
	Field (9/21): SRC_AS
	Field (10/21): DST_AS
	Field (11/21): BGP_NEXT_HOP
	Field (12/21): OUTPUT_SNMP
	Field (13/21): BYTES
	Field (14/21): PKTS
	Field (15/21): FIRST_SWITCHED
	Field (16/21): LAST_SWITCHED
	Field (17/21): IP_NEXT_HOP
	Field (18/21): SRC_MASK
	Field (19/21): DST_MASK
	Field (20/21): TCP_FLAGS
	Field (21/21): IP_PROTOCOL_VERSION
	Type: IP_DST_ADDR (12)
	Type: IP_TOS (5)
	Type: PROTOCOL (4)
	Type: L4_SRC_PORT (7)
	Type: L4_DST_PORT (11)
	Type: ICMP_TYPE (32)
	Type: INPUT_SNMP (10)
	Type: SRC_AS (16)
	Type: DST_AS (17)
	Type: BGP_NEXT_HOP (18)
	Type: OUTPUT_SNMP (14)
	Type: BYTES (1)
	Type: PKTS (2)
	Type: FIRST_SWITCHED (22)
	Type: LAST_SWITCHED (21)
	Type: IP_NEXT_HOP (15)
	Type: SRC_MASK (9)
	Type: DST_MASK (13)
	Type: TCP_FLAGS (6)
	Type: IP_PROTOCOL_VERSION (60)
	Length: 4
	Length: 1
	Length: 1
	Length: 2
	Length: 2
	Length: 2
	Length: 4
	Length: 4
	Length: 4
	Length: 4
	Length: 4
	Length: 4
	Length: 4
	Length: 4
	Length: 4
	Length: 4
	Length: 1
	Length: 1
	Length: 1
	Length: 1

** All length fields are combined, unable to co-relate each other,

Expectation is:

Field (1/21): IP_SRC_ADDR {

                Type: IP_SRC_ADDR (8)
                Length: 4
}

is there a wayout for this?

**Comment by KimiNewt** [comment]
I'm not sure I understand what's the issue. Can you rephrase or post the pcap with expected output?

Perhaps the .alternate_fields property of field might help you?

**Comment by harashee** [comment]
I am facing the same issue. What @srinivasmiskin meant is in CFLOW layer, we get multiple sets of flows. However when we try to parse it using pyshark. That hierarchy is lost. If you see the below pretty_print of cflow layer. There is two flows - **Flow 1** and **Flow 2**  with each having different set of DstAddr and SrcAddr. But here it got two set of values without any hierarchy to traverse properly.

Layer CFLOW:
        FlowSet 1 [id=1024] (2 flows)
        FlowSet Id: (Data) (1024)
        CurrentSecs: 1523013382
        FlowSequence: 260
        Count: 2
        SysUptime: 707.374000000 seconds
        Timestamp: Apr  6, 2018 11:16:22.000000000 UTC
        Version: 9
        Template Frame: 1
        DstAddr: 25.0.2.253 (value overwritten)
        SrcAddr: 25.0.2.73 (value overwritten)
        FlowSet Length: 20
        SourceId: 0
        **Flow 1
        Flow 2**
        DstAddr: 25.0.2.73
        SrcAddr: 25.0.2.253

Please find the attached pcap file where you could see clearly two flows Flow 1 and Flow 2 properly segregated if you open in Wireshark. Please let me know if it is still not clear


[2flow-1.zip](https://github.com/KimiNewt/pyshark/files/1883743/2flow-1.zip)
